### PR TITLE
fix kebab case issue

### LINKED ISSUE #272: Hello world example fails when project name contains underscore
I haven't diagnosed this. But with a new cargo-component installed (v0.11.0), calling `cargo component new hello` builds a cargo directory within which `cargo component build` succeeds but if `cargo component new hello_world` is used, the build step then does not succeed.

```bash
try $ cargo component new hello_world
     Created binary (application) `hello_world` package
     Updated manifest of package `hello_world`
   Generated source file `src/main.rs`
 try $ cd hello_world
 hello_world $ cargo component build
  Generating bindings for hello_world (src/bindings.rs)
   Compiling wit-bindgen-rt v0.24.0
   Compiling bitflags v2.5.0
   Compiling hello_world v0.1.0 (/Users/frank/wasm/try/hello_world)
    Finished dev [unoptimized + debuginfo] target(s) in 0.50s
    Creating component target/wasm32-wasi/debug/hello_world.wasm
error: decoding custom section component-type:wit-bindgen:0.24.0:hello_world:encoded world

Caused by:
    export name `component:hello_world/hello_world` is not a valid extern name
    expected `/` after package name (at offset 0x26)
```

### PR DESCRIPTION
**Pull request by fibonacci1729**
Fixes #272

### CODE REVIEW

File: src/metadata.rs [hunk 1]
```diff
@@ -378,7 +379,7 @@ impl ComponentMetadata {
         }
 
         Ok(Self {
-            name: package.name.clone(),
+            name: package.name.clone().to_kebab_case(),
```
Comment (peterhuene) [inline src/metadata.rs:382]: The package name in the metadata should match the name from the cargo metadata, ideally.

Instead, I think this fix needs to be applied to https://github.com/bytecodealliance/cargo-component/blob/main/src/bindings.rs#L434 (replacing the `clone` call).

**Review by peterhuene — COMMENTED** [review]
Additionally, could we add a build test for a bin project named `hello_world`?

File: src/metadata.rs (hunk 1 above)
Comment (fibonacci1729) [reply src/metadata.rs]: Thanks! Updated and added a build test.

**Review by peterhuene — APPROVED** [review]
Thanks for fixing this!

**Pull request merged**